sdk/src/cameras/3d-smart-camera/calibration_3d_smart.cpp

```diff
@@ -414,3 +422,72 @@ void Calibration3D_Smart::buildGeometryCalibrationCache(
         }
     }
 }
+
+void Calibration3D_Smart::buildDistortionCorrectionCache(unsigned int width,
+                                                         unsigned int height) {
+    using namespace aditof;
+
+    double fx = (double)m_intrinsics[0];
+    double fy = (double)m_intrinsics[4];
+    double cx = (double)m_intrinsics[2];
+    double cy = (double)m_intrinsics[5];
+    std::vector<double> k{m_intrinsics[4], m_intrinsics[5], 0, 0,
+                          m_intrinsics[6]};
+
+    if (m_distortion_cache) {
+        delete[] m_distortion_cache;
+    }
+    m_distortion_cache = new uint32_t[width * height];
+    for (uint16_t i = 0; i < width; i++) {
+        for (uint16_t j = 0; j < height; j++) {
+            double x = (double(i - cx) / fx);
+            double y = double(j - cy) / fy;
+
+            //DISTORTION_COEFFICIENTS for [k1, k2, p1, p2, k3]
+            double r2 = x * x + y * y;
+            double k_calc =
+                (1 + k[0] * r2 + k[1] * r2 * r2 + k[4] * r2 * r2 * r2);
+            //In this case p1 and p2 is always 0
+            //float delta_x = 2 * k[2] * x * y + k[3] * (r2 + 2 * x * x);
+            //float delta_y = k[2] * (r2 + 2 * y*y) + 2 * k[3] * x*y;
+            m_distortion_cache[j * width + i] = k_calc;
+        }
+    }
+}
+
+aditof::Status Calibration3D_Smart::distortionCorrection(uint16_t *frame,
+                                                         unsigned int width,
+                                                         unsigned int height) {
+    using namespace aditof;
+    double fx = (double)m_intrinsics[0];
+    double fy = (double)m_intrinsics[4];
+    double cx = (double)m_intrinsics[2];
+    double cy = (double)m_intrinsics[5];
+    uint16_t *buff = new uint16_t[width * height];
+
+    for (uint16_t i = 0; i < width; i++) {
+        for (uint16_t j = 0; j < height; j++) {
+
+            //transform in adimantional space
```

Comment:
dimensionless because pixel width is divided with pixel size. In other word normalized image coordinates. (https://www.mathworks.com/help/vision/ug/camera-calibration.html)